src/Service/PackageSynchronizer/ComposerPackageSynchronizer.php

```diff
@@ -175,6 +176,13 @@ public function synchronize(Package $package): void
                         $package->addLink(new Link(Uuid::uuid4(), $package, 'suggests', $linkName, $linkDescription));
                         $encounteredLinks['suggests-'.$linkName] = true;
                     }
+
+                    // Is abandoned ?
+                    if ($latest instanceof CompletePackage && $latest->isAbandoned()) {
+                        $package->setReplacementPackage($latest->getReplacementPackage() ?? '');
```

Comment:
this is tricky, so null means normal package, empty string means abandoned package without replacement, and non empty string means abandoned package with replacement